### ISSUE TITLE
False positive RST902 (tuple without redundant parentheses)

### ISSUE THREAD

**Issue by skirpichev** [peterjc/flake8-rst-docstrings#24]
Example:
```

__all__ = 'IPolys',

```

**Comment by peterjc** [comment]
Your report could have been much clearer:

```
$ cat all_tuple.py 
__all__ = 'IPolys',
$ flake8 --select RST all_tuple.py 
all_tuple.py:0:1: RST902 Failed to parse __all__ entry.
```

Versus this version which is black compliant with no error:

```
$ cat all_tuple2.py 
__all__ = ("IPolys",)
$ flake8 --select RST all_tuple2.py
```

This error comes from the code copied from ``pydocstyle/parser.py`` raising ``AllError`` - it could be a problem there too (or if they have fixed it, we can back port their fix).

**Comment by webknjaz** [comment]
Just hit the same bug. Subscribing to updates.

**Comment by webknjaz** [comment]
Also, FTR, the trailing comma is not what's causing the parse error. I've hit this on the following line:
```python
__all__ = 'signature', 'wraps'  # noqa: WPS410  # explicit re-export
```

**Comment by peterjc** [comment]
Looks like https://github.com/PyCQA/pydocstyle/pull/355 fixed this upstream, but enough has changed in ``src/pydocstyle/parser.py`` that the backport is not trivial. Their new ``is_surrounded`` boolean enables support for multiple line ``__all__`` statements too.

**Comment by peterjc** [comment]
I have a potential fix in #33 (which fixes a long term issue), but will likely raise the minimum version of Python and/or pydocstyle required. If any of you are able to help test that (and combinations of the dependencies) that would be great.

**Comment by peterjc** [comment]
I'm now fairly certain that the change on #33 (importing ``pydocstyle`` v3 onwards which handles ``__all__`` better) would also mean dropping Python 2 support.

Based on https://pypistats.org/packages/flake8-rst-docstrings the Python 2 downloads are well under 1%, and we're well past the Python 2 sunset, so I'd be OK with that.

If we really wanted to preserve Python 2 support, that would mean sticking with our bundled obsolete copy of the pydocstyle parser, but editing it to apply the logic of their ``__all__`` fix. A pain but it ought to be doable.

Any comments?

**Comment by peterjc** [comment]
This will be resolved in the next release, v0.1.0, which also drops Python 2 support.

**Comment by webknjaz** [comment]
@peterjc thanks for taking care of this! Any reason you don't publish wheels?

**Comment by peterjc** [comment]
No wheels was an oversight (although does it actually cause much trouble with a pure Python package?), addressed with your prompting on #34.

**Comment by webknjaz** [comment]
> although does it actually cause much trouble with a pure Python package

Mostly annoyance: 1) wheels are installed _much_ faster than tarballs; 2) wheels add more complete metadata to PyPI which affects how the dependency resolvers work — when the pip's resolver encounters a tarball-only dependency, it has to download and build it, sometimes multiple versions of the same distribution if it ends up doing backtracking, each of such cases affects the overall `pip install` experience.
